Code Review

awhitty1 opened the pull request:
Hello Luis,
 
Can you please review my code for the trigrams exercise. I'm not sure if the random data from the list is being generated.
(I know it's late to be asking...just got home from work.)

ldconejo commented on students/alex_whitty/lesson4/trigrams.py:
```diff
@@ -0,0 +1,83 @@
+            ("Space", "dug"): ["into"],
+            ("dug", "into"): ["the"],
+            ("into", "the"): ["pile"],
+            }
```
Remove this dictionary to avoid confusion, as the code is already capable of generating it's own trigram dict.

ldconejo commented on students/alex_whitty/lesson4/trigrams.py:
```diff
@@ -0,0 +1,83 @@
+# build up the dict here!
+
+def build_dict(words):
+    starter = random.choice(list(words.keys("from", "Outer")))
```
For this one, you don't need to add ("from", "Outer"), the idea is that starter will use a random key from _words_

ldconejo commented on students/alex_whitty/lesson4/trigrams.py:
```diff
@@ -0,0 +1,83 @@
+    # Form the next round, using the second
+    # word of the starter and next_word
+    # and repeat the process
+
```
You need to continue the sequence:
1. Create a new tuple key, using the second word in _starter_ and _next_word_.
2. Get the list of words associated with the new key.
3. Select a random word from that list.
4. Add that word to _new_text_
5. Repeat steps 1-4 until a predefined limit, for example 1000 iterations (so that you get a new text with 1000 words).

ldconejo commented on students/alex_whitty/lesson4/trigrams.py:
```diff
@@ -0,0 +1,83 @@
+
+
+if __name__ == "__main__":
+    words = "When the Travelers from outer Space dug into the pile".split()
```
You should update this one to call _parse_file()_ to produce your list of words.

ldconejo commented:
Hi Alex,

I added a few comments to your code. You still need to complete _build_dict()_ so that it continues adding words to the new text, I added some comments on the sequence you need to follow for that (you can implement it with a _while_ loop).

Please feel free to submit a new pull request once you have a new version of your code ready for review.

Regards,

Luis

awhitty1 commented:
Thanks Luis for the assistance.

> On Feb 10, 2019, at 6:34 PM, Luis Conejo-Alpizar <notifications@github.com> wrote:
> 
> Hi Alex,
> 
> I added a few comments to your code. You still need to complete build_dict() so that it continues adding words to the new text, I added some comments on the sequence you need to follow for that (you can implement it with a while loop).
> 
> Please feel free to submit a new pull request once you have a new version of your code ready for review.
> 
> Regards,
> 
> Luis
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/UWPCE-PythonCert-ClassRepos/GP_Python210B_Winter_2019/pull/84#issuecomment-462206180>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AlDPosKDQIoeiXLFlf8eK5SMvtSRkq1oks5vMNbAgaJpZM4azZ00>.
>
